### ISSUE TITLE
Add ability to download pre-built indexes

### ISSUE THREAD

**Issue by lintool** [castorini/anserini#2296]
This is a feature from Pyserini, but it'd be great to have in Anserini also - something like:

```
target/appassembler/bin/SearchCollection \
  -index msmarco-v1-passage \
  -topics tools/topics-and-qrels/topics.msmarco-passage.dev-subset.txt \
  -topicReader TsvInt \
  -output runs/run.msmarco-passage.bm25-default.topics.msmarco-passage.dev-subset.txt \
  -bm25
```

Use the same identifiers as in https://github.com/castorini/pyserini/blob/master/pyserini/prebuilt_index_info.py#L167 ?

Bunch of design choices we need to make as we implement, e.g.,

+ download to `~/.cache/pyserini/indexes` or create new `~/.cache/anserini/indexes`?
+ use `prebuilt_index_info.py` (i.e., read from that file...) or make a copy in Anserini? (of the relevant subset)

**Comment by lintool** [comment]
I poked around a bit, and this seems to be the most popular implementation of Java progress bars:

https://tongfei.me/progressbar/styles/

Gist for a downloader:
https://gist.github.com/AlonEirew/427128bacee1f67b1428031730293f13

**Comment by lintool** [comment]
Ref: #2301

**Comment by lintool** [comment]
Closed by #2301 🎉
